### PR TITLE
FIX: Ensure closing sidebar tears down all callbacks.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/sidebar/common/community-section.js [hunk 1]
```diff
@@ -35,7 +35,16 @@ export default class SidebarCommunitySection extends Component {
...
 
   willDestroy() {
-    this.sectionLinks.forEach((sectionLink) => sectionLink.teardown());
+    [
+      ...this.sectionLinks,
+      ...this.moreSectionLinks,
+      ...this.moreSecondarySectionLinks,
+    ].forEach((sectionLink) => {
+      if (sectionLink.teardown) {
+        sectionLink.teardown();
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/components/sidebar/common/community-section.js:44]: minor not: I know in chat we've been doing this form:

```javascript
this.onChangeSelectedUsers?.(this.selectedUsers);
```

So maybe we could do that here for consistency?

```javascript
sectionLink.teardown?.()
```

**Review by martin-brennan — APPROVED** [review]
Nice, only a minor nit but don't need to change necessarily

File: app/assets/javascripts/discourse/app/components/sidebar/common/community-section.js (hunk 1 above)
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/components/sidebar/common/community-section.js]: O this is nicer :) I've updated the code

**Comment by tgxworld** [comment]
Thank you @martin-brennan

**Pull request merged**